nemo/collections/tts/data/datalayers.py

```diff
@@ -117,3 +124,73 @@ def __getitem__(self, index):
 
     def __len__(self):
         return len(self.collection)
+
+
+@experimental
+class AudioToPhonemesDataset(_AudioTextDataset):
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+               """
+        return {
+            "audio_signal": NeuralType(
+                ("B", "T"),
+                AudioSignal(freq=self._sample_rate)
+                if self is not None and hasattr(self, "_sample_rate")
+                else AudioSignal(),
+            ),
+            "a_sig_length": NeuralType(tuple("B"), LengthsType()),
+            "transcripts": NeuralType(("B", "T"), LabelsType()),
+            "transcript_length": NeuralType(tuple("B"), LengthsType()),
+        }
+
+    def __init__(
+        self,
+        manifest_filepath: str,
+        cmu_dict_path: str,
+        sample_rate: int,
+        int_values: bool = False,
+        augmentor: "nemo.collections.asr.parts.perturb.AudioAugmentor" = None,

```

Comment:
1) Can you justify need for str type marking here?
2) [Here and below] Why do you type args in both docstring and method declaration? For example, this is strongly discourage in Google's style guide.

Reply:
1. Changed it, though this is how it is in the asr collection datalayers
2. Changed to method